rule.go

```diff
@@ -220,22 +229,22 @@ func escape(r string) string {
 }
 
 // within returns the within value for a specific content.
-func within(options []*ContentOption) int {
+func within(options []*ContentOption) string {
 	for _, o := range options {
 		if o.Name == "within" {
-			return int(o.Value)
+			return o.Value
 		}
 	}
-	return 0
+	return ""
 }
 
 // RE returns all content matches as a single and simple regexp.
 func (r *Rule) RE() string {
 	var re string
 	for _, c := range r.Contents {
 		// TODO: handle pcre, depth, offset, distance.
-		if w := within(c.Options); w != 0 {
-			re += fmt.Sprintf(".{0,%d}", w)
+		if w := within(c.Options); w != "" {
+			re += fmt.Sprintf(".{0,%s}", w)
```

Comment:
This actually does expect an int, and won't work correctly with things that use variables in this position.
The idea for this function is to allow generation of (a relatively simple) regexp that can be applied to other log sources, etc. based on a content match.
If the value in w is a variable name, this will break.

can you add an Atoi check and keep this as %d ?

Reply:
done